### PR TITLE
New ServerGrabber

### PR DESCRIPTION
**Pull request by Nicogene**
This PR introduces the new `ServerGrabber`. 
For now can be instantiated with `--device grabberDual`, then , when the old server will be removed, simply with `--device grabber`.
These are the features of the new server: 

- it removes the inheritance from some "old" classes and interfaces:   
``` c++
IFrameGrabberImage,
IAudioGrabberSound,
IAudioVisualGrabber,
IFrameGrabberControls,
IService,
DataSource<yarp::sig::ImageOf<yarp::sig::PixelRgb> >,
DataSource<yarp::sig::ImageOf<yarp::sig::PixelMono> >,
DataSource<yarp::sig::Sound>,
DataSource<ImageRgbSound>,
DataSource2<yarp::sig::ImageOf<yarp::sig::PixelRgb>,yarp::sig::Sound>
```
- it inherits from interfaces like `public yarp::dev::IWrapper` and public `yarp::dev::IMultipleWrapper`, that allow to run the server through `yarprobotinterface`
- it has rpc(`/grabber/rpc`) and streaming ports separated
- it allows you to decide to open one or two cameras, RGB or RAW, and in case of two cameras you can decide to send the images in two separate ports(`/grabber/left` and (`/grabber/right`) or in one(`/grabber`) with the two images stitched horizontally
- it lost the audio support for now, if needed we will introduce it later

**TODO** brefore merging it  :

- [x]  Test launching it with the `yarprobotinterface`
- [x]  Write yarp tests
- [x]  Merge PR #1106 and rebase
- [x]  Write documentation 
- [x]  make `setExternal` work with dragonfly
- [x]  add the second rpc port, in case you use two cameras.
- [x]  Test with the dragonFly of the robot
- [x]  Increase the yarp tweak number to 7
- [x]  Update `doc/release/v2_3_70.md`

Tests on iCubGenova01 passed.
Please merge :+1:

### CODE REVIEW

**Comment by lornat75** [comment]
+1 especially for the option that allows streaming stereo pairs in a single image. However I do not see a strong reason for moving the grabbers inside yarprobotinterface.

File: src/libYARP_dev/src/devices/ServerGrabber/ServerGrabber.cpp [hunk 1]
```diff
@@ -888,10 +888,10 @@ void ServerGrabber::run()
                     flex_i2.setQuantum(i2.getQuantum());
                     flex_i2.resize(fgImage2->width(),fgImage2->height());
                     //ASK ma flex_i.width e height saranno 0 giusto?
-                    i.setExternal(flex_i.getRawImage(),flex_i.width(),flex_i.height());
                     fgImage->getImage(i);
-                    i2.setExternal(flex_i2.getRawImage(),flex_i2.width(),flex_i2.height());
                     fgImage2->getImage(i2);
+		    memcpy(flex_i.getRawImage(), i.getRawImage(), i.getRawImageSize());
+		    memcpy(flex_i2.getRawImage(), i2.getRawImage(), i2.getRawImageSize());
```
Comment (traversaro) [inline src/libYARP_dev/src/devices/ServerGrabber/ServerGrabber.cpp]: ⚠️️⚠️️ Tabs alert! ⚠️️⚠️️

Comment (barbalberto) [inline src/libYARP_dev/src/devices/ServerGrabber/ServerGrabber.cpp]: Yes, don't worry. this commit is temporary (done from the robot).
It is gonna be fixed and squashed from a pc with a decent editor

**Comment by barbalberto** [comment]
@lornat75 Sure, there is no reason to change the status quo.
We want to check that the 2 ways of instantiating the server/camera(s) works: either using subdevice keyword or using `attachAll()` . Then the user can choose.

We tested this device and it is also 100% back compatible with the previous one _**for image streaming**_.

Right now the new `grabberDual` does not support audio streaming.
We may think of introducing this feature later on...

**Comment by Nicogene** [comment]
We added the additional rpc port for handling the two cameras separately, tested on the robot successfully. 
We added the application `Calib_DualCameras` on `iCubGenova01`. it launches the new `grabberDual` with the images split in two ports.
Please try to use it instead of `Calib_Cameras`, and report to me or to @barbalberto any problem(devel on the robot has to be updated).

**Comment by barbalberto** [comment]
All changes requiested have been applied by @Nicogene. The parameters for the 2 cameras can be changed independetly by mean of 2 rpc ports.

We are planning to merge this PR on monday. If other changes shall be done, please comment.
@lornat75  @drdanz

P.S.
The use of 'main' thread when started with yarpdev is gonna be done in a separated PR as an improvement.

File: data/CMakeLists.txt [hunk 2]
```diff
@@ -4,4 +4,5 @@
 
 add_subdirectory(yarpmanager)
 add_subdirectory(yarpdataplayer)
+add_subdirectory(harness_dev)
```
Comment (drdanz) [inline data/CMakeLists.txt]: This should not be here

Comment (Nicogene) [inline data/CMakeLists.txt]: Hi @drdanz, 
we added this directory containing the ini files for running `TestFrameGrabberTest`, that needs them to use the new `ServerGrabber`.
We created this context for `harness_dev`, where should it be ?

Comment (drdanz) [inline data/CMakeLists.txt]: Somewhere in tests... And it should not be installed

Comment (Nicogene) [inline data/CMakeLists.txt]: @drdanz I have extended `YARP_DATA_DIRS` in `harness.cpp` as you suggested me, adding a new folder inside `tests` called `conf` with the configuration files required by `TestFrameGrabberTest`.
All the changes have been squashed in e88c130 rebasing over `origin/devel`.
Is it ok for you ?

**Comment by Nicogene** [comment]
@drdanz I created the context as you asked me :point_right: bbe002a549b9324beb49b0ed7e925acc73a53b0e
I have already rebased over origin/devel.

**Comment by barbalberto** [comment]
Suggested changes have been applied. Rebased and merged.

**Pull request merged**